Calculating

tsohleDev opened the pull request:
**Connecting the components**

- Use props to send the calculation string from the parent state
- lifted the state from the buttons in order to make the calculations and send them to the screen

tsohleDev committed:
Add calculating logic

tsohleDev committed:
Impliment the logic to the calculator
Use props and state and state up

codepantha commented on src/Components/Calculator.js:
```diff
@@ -1,16 +1,38 @@
+        <Screen total={total} next={next} operation={operation} className="screen" />
+        {buttons.map((button) => (
+          <Button
+            key={button.id}
```
- [x] The `id` attribute does not exist on the button item. This is why you have this console error asking for a "unique" key prop shown below 👇 Kindly do `key={button}` instead 👍 
![image](https://user-images.githubusercontent.com/46839548/212122251-57a748f9-04d6-4a22-8a40-c07fa4324289.png)

codepantha commented on src/Components/Screen.js:
```diff
@@ -1,13 +1,34 @@
   }
 }
 
+Screen.propTypes = {
+  total: PropTypes.string.isRequired,
+  next: PropTypes.string.isRequired,
+  operation: PropTypes.string.isRequired,
+};
```
- [x] There are about 3 console errors which I'll show below 👇 as a result of your PropTypes.
![image](https://user-images.githubusercontent.com/46839548/212124985-646c8744-cf96-4187-a2b9-d4d436ec13a9.png)
1. the `total` prop should be a number and not a string, so please specify `PropTypes.number.isRequired`
2. when the page first loads and you have not begun any arithmetic operation, the value of `next` prop is `null` so specifying in your PropTypes that `next` is required will be problematic. Please specify `PropTypes.string` only.
3.  same as number 2 above. Please specify `operation` prop to be `PropTypes.string` only

Having done that, you will notice that you need a `defaultProps` declaration. Kindly do this 👇 
```javascript
Screen.defaultProps = {
  next: null,
  operation: null,
};
```
Please visit this [link](https://www.geeksforgeeks.org/reactjs-defaultprops/) if you need to research more on React defaultProps 😎

codepantha reviewed (changes requested):
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @tsohleDev 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

<img src="https://media2.giphy.com/media/xUOxfh6ZM75efM3Bqo/giphy-downsized-medium.gif"/>

## Highlights 🎯 
✔️ Descriptive PR
✔️ Added events and functionality to calculator app 📌 
✔️ Good commit history
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

tsohleDev committed:
Remove Console errors
Manifest json: delete icon references
Prop types: set default props
unique id for buttons

Ismailco reviewed (approved):
### Hi @tsohleDev, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

![cr-approve](https://user-images.githubusercontent.com/25227022/165483662-7b06c812-4cc0-40b9-a84c-4729d4cf10c3.gif)


## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
